hw7_functions/functional.py

```diff
@@ -0,0 +1,64 @@
+# Functional programming
+def sequential_map(*args):
+    """
+    Sequentially applies functions to container values
+    :param args: functions, a container
+    :return: result of sequential application of functions to container values
+    """
+    *functions, container = args
+    for function in functions:
+        container = list(map(function, container))
+    return container

```

Comment:
Превратить элемент в список можно только в самом конце, результат map можно передавать в другие map'ы
```suggestion
        container = map(function, container)
    return list(container)

```